### ISSUE TITLE
Empty string in access Groups breaks data access control 

### ISSUE THREAD

**Issue by LAShemilt** [SciCatProject/backend#1193]
## Issue Name
Bug - empty string in access Groups breaks data access control 

### Summary
When a User Identity is created from an OIDC login, there is always an empty string appended to the list of the user's access groups in the mongo db (`db.UserIdentity`). This creates an access group called `""`

When a Dataset is uploaded with no access groups , and only an owner group, the access groups for this dataset is created as an empty string in the `db.Dataset` entry.

This means that all users can see this dataset, because they all belong to the access group `""`. 

### Steps to Reproduce
We have a local instance of SciCat that is linked to our dev Keycloak server for OIDC authentication. 
We have the following variables configured:
```
OIDC_ACCESS_GROUPS_PROPERTY: groups
ACCESS_GROUPS_OIDCPAYLOAD_ENABLED: ${ACCESS_GROUPS_OIDCPAYLOAD_ENABLED} #this value needs to be set to true in a env file as a string "true" is not recognised
ACCESS_GROUPS_API_URL: https://ourdevkeycloak.org/realms/dev
 ACCESS_GROUPS_SERVICE_TOKEN: ${OIDC_CLIENT_SECRET}
```
In our dev keycloak we have two groups `scicat-data-1` and `scicat-data-2`, all users belong to at least one group.

We ingest data using the functionalAccount ingestor, with the following ownable:
```
    {
        "owner": "Tricia McMillan",
        "ownerGroup": "scicat-data-2",
        "accessGroups": ["scicat-data-1"],
        "datasetName": "Dataset 26",

        },
    {
        "owner": "Ford Prefect",
        "ownerGroup": "scicat-data-1",
        "accessGroups": [],
        "datasetName": "Dataset 27",
        },
        {
        "owner": " Marvin",
        "ownerGroup": "scicat-data-2",
        "accessGroups": [],
        "datasetName": "Dataset 28",

        },
```
All users can see all datasets, regardless of if they are not in one of the groups. 

### Current Behaviour
If you uses the set up above, all data can be seen by all users regardless of what groups they  have. When looking at the dataset entry for the datasets that have no access groups they have a  `accessGroup: ""`.

When looking into the `db.UserIdentity` entry for the users we can see that they also have an accessGroup `""`. This does not seem to be brought in from the OIDC payload , but rather added by SciCat. 

```

{ "_id" : ObjectId("XXXXX"),
"authStrategy" : "oidc",
 "externalId" : "XXX", 
"profile" : { "displayName" : "Tricia McMillan", 
"email" : "rfitrillian@rfi.ac.uk", 
"username" : "rfitrillian", 
"thumbnailPhoto" : "no photo",
 "id" : "XXXXDX", 
"emails" : [ { "value" : "rfitrillian@rfi.ac.uk" } ], 
"accessGroups" : [ "", "scicat-data-1", "scicat-data-2" ], 
"oidcClaims" : { "exp" : XXXXXX, "iat" : XXXXXX, "auth_time" : XXXXXX, "jti" : "XXXX", "iss" : "https://ourdevkeycloak.org/realm/dev", "aud" : "scicat-local", "sub" : "XXXXX", "typ" : "ID", "azp" : "scicat-local", "session_state" : "XXXXXX", "at_hash" : "XXXX", "acr" : "1", "sid" : "XXXXX", "accessgroups" : [ "scicat-data-1", "scicat-data-2" ], "email_verified" : false, "name" : "Tricia McMillan", "preferred_username" : "rfitrillian", "given_name" : "Tricia", "family_name" : "McMillan", "email" : "rfitrillian@rfi.ac.uk" }, "_id" : ObjectId("XXXXX") }, "provider" : "https://ourdevkeycloak.org/realm/dev", "userId" : ObjectId("XXXXXXX"), "created" : ISODate("2024-04-26T14:42:28.355Z"), "modified" : ISODate("2024-04-26T14:42:28.355Z"), "__v" : 0 }
```
### Expected Behaviour
If a dataset was not given an access group, I would expect that only users in the Owner Group to see it. When a UserIdentity gets created I would expect it just to have the access groups from the OIDC and not have an empty group attached.

**Comment by bpedersen2** [comment]
Hmm, I don't observe it in our system, maybe check the access_groups_static value in the config, if it is (or contains) an empty string. We have some values configured and they end up (correctly) in the final accessgroups.

**Comment by Junjiequan** [comment]
For the first issue - Empty string in user's `accessGroup`, can you try to include a ENV variable `ACCESS_GROUPS_STATIC_ENABLED=false` and then login again? There's a small logical deficiency with static group values mapping function, I will fix it as soon as possible.

Second issue -  Creates dataset without `accessGroup` and only `ownerGroup` results in new dataset with empty string value accessGroup.
I was not able to reproduce the result. If the accessGroup field is not provided, created dataset only contains empty array `[]` accessGroup field without any values. Could you provide the dataset JSON value that you've used for creating the dataset?

**Comment by LAShemilt** [comment]
Thanks for getting back. I can confirm that by adding `ACCESS_GROUPS_STATIC_ENABLED=false` to the environment vars I was able to get rid of the empty string.

The dataset I am uploading via curl has the following access groups var:
```
{
        "owner": "Ford Prefect",
        "ownerGroup": "scicat-data-1",
        "accessGroups": [],
        "datasetName": "Dataset 27",
        },
} 
```
Where access groups are given by an empty list. I am currently ingesting with the `requests` package in python (curl equivalent).
If there is no access group should the value be something different?

**Comment by nitrosx** [comment]
If there are no `accessGroups` for the dataset, assigning the value [] is correct.

**Comment by nitrosx** [comment]
Closing this issue as it should be solved with PR #1203.
We should add tests for the use case reported in this issue. Will add a new issue.
